### PR TITLE
CLOUDP-268810: Fixed IDP custom roles e2e test flakiness

### PR DESCRIPTION
**Pull request by igor-karpukhin**
### All Submissions:

Fixed IDP custom roles e2e test flakiness

* [x] Have you signed our [CLA](https://www.mongodb.com/legal/contributor-agreement)?
* [x] Put `closes #XXXX` in your comment to auto-close the issue that your PR fixes (if there is one).
* [ ] Update docs/release-notes/release-notes-template.md if your changes should be included in the release notes for the next release.

### CODE REVIEW

**Comment by helderjs** [comment]
@igor-karpukhin this seems to contain changes from the previous PR

**Comment by igor-karpukhin** [comment]
> @igor-karpukhin this seems to contain changes from the previous PR

Yes, I think we're going to merge the Helm fix first :D

**Review by josvazg — APPROVED** [review]

File: test/e2e/independent_customroles_test.go [hunk 1]
```diff
@@ -134,9 +134,11 @@ var _ = Describe("Migrate one CustomRole from AtlasProject to AtlasCustomRole re
...
 
 		By("Enabled reconciliation for AtlasProject", func() {
-			Expect(testData.K8SClient.Get(testData.Context, client.ObjectKeyFromObject(testData.Project), testData.Project)).Should(Succeed())
-			testData.Project.Annotations = map[string]string{}
-			Expect(testData.K8SClient.Update(testData.Context, testData.Project)).To(Succeed())
+			Eventually(func(g Gomega) {
+				g.Expect(testData.K8SClient.Get(testData.Context, client.ObjectKeyFromObject(testData.Project), testData.Project)).Should(Succeed())
+				testData.Project.Annotations = map[string]string{}
+				g.Expect(testData.K8SClient.Update(testData.Context, testData.Project)).To(Succeed())
+			}).WithTimeout(1 * time.Minute).WithPolling(5 * time.Second).Should(Succeed())
```
Comment (josvazg) [inline test/e2e/independent_customroles_test.go:141]: Still I think @s-urbaniak had an update implementation to precisely avoid race condition conflicts on updated

Comment (josvazg) [reply test/e2e/independent_customroles_test.go:141]: yep, see [RetryUpdateOnConflict](https://github.com/mongodb/mongodb-atlas-kubernetes/blob/b6bbb6db6ce1c81b546f30babea890f5e5bc1554/test/helper/retry/retry.go#L17)

And also see [sample usage](https://github.com/mongodb/mongodb-atlas-kubernetes/blob/b6bbb6db6ce1c81b546f30babea890f5e5bc1554/test/e2e/backup_compliance_test.go#L145)

**Review by josvazg — CHANGES_REQUESTED** [review]
Prefer [RetryUpdateOnConflict](https://github.com/mongodb/mongodb-atlas-kubernetes/blob/b6bbb6db6ce1c81b546f30babea890f5e5bc1554/test/helper/retry/retry.go#L17)

File: test/e2e/independent_customroles_test.go (hunk 1 above)
Comment (s-urbaniak) [reply test/e2e/independent_customroles_test.go:141]: yeah, let's use the canonical one.

Comment (igor-karpukhin) [reply test/e2e/independent_customroles_test.go:141]: Why is it better? I prefer the explicit way of doing such checks to be honest.

Comment (josvazg) [reply test/e2e/independent_customroles_test.go:141]: This is not only also explicit, but less error prone, just a single function call:
```go
			_, err := akoretry.RetryUpdateOnConflict(
				ctx,
				testData.K8SClient,
				client.ObjectKeyFromObject(...),
				func(...) {...})
			Expect(err).To(BeNil())
```

Comment (s-urbaniak) [reply test/e2e/independent_customroles_test.go:141]: @igor-karpukhin i highly suggest using the central function, it is a small wrapper around the central client-go `RetryUpdateOnConflictFunction`, see https://github.com/kubernetes/client-go/blob/master/util/retry/util.go#L68-L101.

This function is battle-tested throughout Kubernetes and used in many operators, it includes backoff and considers only Conflict errors which both things are not handled by your function.

**Comment by igor-karpukhin** [comment]
@josvazg do you see this as a blocker, or can we proceed and merge this PR?

**Comment by josvazg** [comment]
I highly suggest using **RetryUpdateOnConflict** or we may patch it later again anyway

**Review by s-urbaniak — APPROVED** [review]

**Review by josvazg — APPROVED** [review]

**Pull request merged**